com.unity.multiplayer.mlapi/Tests/Runtime/BufferDataValidationComponent.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using MLAPI.Messaging;
+
+namespace MLAPI.RuntimeTests
+{
+    /// <summary>
+    /// Used in conjunction with the RpcQueueTest to validate from 1 byte to (n) MaximumBufferSize
+    /// - Sending and Receiving a continually growing buffer up to (MaximumBufferSize)
+    /// - Default maximum buffer size is 1MB
+    /// </summary>
+    public class BufferDataValidationComponent : NetworkBehaviour
+    {
+        /// <summary>
+        /// Allows the external RPCQueueTest to begin testing or stop it
+        /// </summary>
+        public bool EnableTesting;
+
+        /// <summary>
+        /// The maximum size of the buffer to send
+        /// </summary>
+        public int MaximumBufferSize = 1 << 20;
+
+        /// <summary>
+        /// The rate at which the buffer size increases until it reaches MaximumBufferSize
+        /// (the default starting buffer size is 1 bytes)
+        /// </summary>
+        public int BufferSizeStart = 1;
+
+        /// <summary>
+        /// Is checked to determine if the test exited because it failed
+        /// </summary>
+        public bool TestFailed { get; internal set; }
```

Comment:
nit-pick:
```suggestion
        public bool IsTestFailed { get; internal set; }
```

Reply:
I think TestFailed is fine.  IsTestFailed doesn't sound right.  :)
Maybe DidTestFail would sound right, but TestFailed is easy to understand.